Add Workspace Members Search Feature

PankajAS opened the pull request:
### Details

### Fixed Issues
$ https://github.com/Expensify/App/issues/12339
PROPOSAL: [#12339(comment)](https://github.com/Expensify/App/issues/12339#issuecomment-1422805292)


### Tests
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

- [] Verify that no errors appear in the JS console

### Offline tests
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

### QA Steps
1. Navigate to Settings.
2. Click on Workspaces.
3. Click on any Workspace and Manage members.
4. Search member by name or email address.

- [] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/17126628/218137774-ba89c425-880c-4c65-a526-b91aa097b2aa.mp4



</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/17126628/218143163-82d0d273-dfb8-488d-972a-3d6d57eac7b1.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/17126628/218138191-8e2704d8-61ab-4cec-abdd-4bf43b4f5bc4.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/17126628/218138700-628d7aca-7f51-4c78-a849-0e940c521af6.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/17126628/218139516-f704b589-0fbb-4288-965e-d8c02c791062.mp4


</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/17126628/218138792-fc8b5061-99bc-462a-80a4-305ae6d91663.mov

</details>

PankajAS committed:
Add Search feature for Workspace members list

1. Added common isKeywordMatch function
2. Add more fields in filter condition: firstName, lastName
3. Combine all of these into one if statement
4. Use common function like clear()

1. Put this.state.searchValue.trim().toLowerCase() this out of loop
2. Space formate

UI Adjustments

Match "No results found" text with new chat text

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

1. Remove updateSearchValue Comment
2. Update OnSearch name to updateSearchValue
3. Inline conditions for Filter list data

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Miroslav Stevanovic <97473779+0xmiroslav@users.noreply.github.com>

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

PankajAS commented:
@0xmiroslav i have created new PR please check

PankajAS commented:
@0xmiroslav I have changed according to these feedbacks:
1. Added common isKeywordMatch function
2. Add more fields in filter condition: firstName, lastName
3. Combine all of these into one if statement
4. Use common function like clear()

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -299,6 +327,19 @@ class WorkspaceMembersPage extends React.Component {
+        data = _.filter(data, (member) => {
+            const searchValue = this.state.searchValue.trim().toLowerCase();
+
+            if (this.isKeywordMatch(member.displayName,searchValue)
```
do you mean by space?

PankajAS replied:
Done

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -299,6 +327,19 @@ class WorkspaceMembersPage extends React.Component {
         });
         data = _.sortBy(data, value => value.displayName.toLowerCase());
+        data = _.filter(data, (member) => {
+            const searchValue = this.state.searchValue.trim().toLowerCase();
```
Done

shawnborton commented:
This looks pretty good to me. I wonder if we can tighten up some of the margins though to take advantage of the space better? Basically remove some pixels from the places I highlighted below:

<img width="729" alt="image" src="https://user-images.githubusercontent.com/2319350/218737106-caaa75d0-ec50-416c-b808-de7c6cd45b2b.png">

PankajAS commented:
@0xmiroslav @shawnborton is thik looks ok now?
<img width="389" alt="Screenshot 2023-02-14 at 6 01 57 PM" src="https://user-images.githubusercontent.com/17126628/218739777-fedbb101-51ef-480f-992c-d252d5a34d6d.png">

shawnborton commented:
Yeah, agree - just want to agree that you followed the guidance I provided before signing off. It looks like maybe more than 4 was removed from above the search input?

PankajAS commented:
> Yeah, agree - just want to agree that you followed the guidance I provided before signing off. It looks like maybe more than 4 was removed from above the search input?

I have just remove 4px @shawnborton as i set styles.pt1 in place of styles.pt5

shawnborton commented:
Ah, well the difference petween pt1 (4px) and pt5 (20px) is actually 16. So really you would want to use pt4 instead of pt5.

PankajAS commented:
> Ah, well the difference between pt1 (4px) and pt5 (20px) is actually 16. So really you would want to use pt4 instead of pt5.

Ohk @shawnborton

PankajAS commented:
@shawnborton looks ok now?

<img width="376" alt="Screenshot 2023-02-14 at 6 32 44 PM" src="https://user-images.githubusercontent.com/17126628/218746689-3c39b6b4-f625-4328-b6b2-82ca12ef31df.png">

PankajAS commented:
> @PankajAS since you're new, maybe you are not familiar with our style guidelines. Styles are defined in styles.js and 1 difference is 4px. i.e.
> 
> ```
>     pt2: {
>         paddingTop: 8,
>     },
> 
>     pt3: {
>         paddingTop: 12,
>     },
> 
>     pt4: {
>         paddingTop: 16,
>     },
> 
>     pt5: {
>         paddingTop: 20,
>     },
> ```
> 
> You can use spacing already defined in /styles/utilities/spacing.js

yes i'm using these styles @0xmiroslav

PankajAS commented:
<img width="375" alt="Screenshot 2023-02-14 at 6 37 25 PM" src="https://user-images.githubusercontent.com/17126628/218747744-7d5397b9-5d07-40c5-bbd8-c536b54f2213.png">


> @PankajAS and to be clear, you only removed 4 from below the search input as well?

yes @shawnborton i remove 4px only from below and top of search input

shawnborton commented:
Cool, thank you. That works for me then!

PankajAS commented:
> Cool, thank you. That works for me then!

Great, let me commit this change

PankajAS commented:
@0xmiroslav i have committed UI adjustment changes feedback, please check

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -391,7 +391,7 @@ class WorkspaceMembersPage extends React.Component {
                                 />
                             </View>
-                            <View style={[styles.w100, styles.p5]}>
+                            <View style={[styles.w100, styles.pv4,styles.ph5]}>
```
Done @0xmiroslav

PankajAS commented:
> @PankajAS please update Screenshots section with latest changes

Ok @0xmiroslav

shawnborton commented:
For no results found, we should match the style of the new chat page.

For auto focus, I don't feel strongly here. Part of me thinks we don't need to auto focus in this particular scenario but I can see where it is consistent.

shawnborton commented:
Works for me!

PankajAS commented:
> @PankajAS please work on [last feedback](https://github.com/Expensify/App/pull/15123#issuecomment-1430118296)

All 4 feedbacks need to implement? @0xmiroslav

PankajAS commented:
> > > @PankajAS please work on [last feedback](https://github.com/Expensify/App/pull/15123#issuecomment-1430118296)
> > 
> > 
> > All 4 feedbacks need to implement? @0xmiroslav
> 
> No, only 3 and 4 for now

Ok @0xmiroslav

PankajAS commented:
<img width="376" alt="Screenshot 2023-02-14 at 11 40 27 PM" src="https://user-images.githubusercontent.com/17126628/218823832-0bf6e202-f3c8-4147-84b9-2c85e0701bae.png">
yes its same as Group chat @0xmiroslav

PankajAS commented:
check now @0xmiroslav

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -21,17 +21,18 @@ import Text from '../../components/Text';
 import ConfirmModal from '../../components/ConfirmModal';
 import personalDetailsPropType from '../personalDetailsPropType';
-import withWindowDimensions, {windowDimensionsPropTypes} from '../../components/withWindowDimensions';
+import withWindowDimensions, { windowDimensionsPropTypes } from '../../components/withWindowDimensions';
```
Ok @0xmiroslav

PankajAS commented:
`I have read the CLA Document and I hereby sign the CLA`

PankajAS commented:
Done @0xmiroslav

PankajAS commented:
> @PankajAS please fix lint errors `npm run lint`



> @PankajAS please fix lint errors `npm run lint`

Fixed @0xmiroslav

0xmiroslav reviewed (changes requested):
requesting minor changes.
After this fix, I think we are good to go

PankajAS commented:
@0xmiroslav these feedbacks done
1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -1,9 +1,7 @@
-import {
-    View, FlatList, TouchableOpacity,
-} from 'react-native';
+import {View, FlatList, TouchableOpacity} from 'react-native';
```
Can you remove any changes like this that are unnecessary please? I will review again once there are only essential changes. If you are using any auto-lint tools please disable them.

marcaaron replied:
Not too concerned with it / don't have any real opinion on it. I'd say it's fine, but not really needed.

PankajAS replied:
@marcaaron @0xmiroslav do i need to revert lint fixes?

PankajAS replied:
@marcaaron i fix this after run `npm run lint`

PankajAS replied:
@0xmiroslav i have fix it please check

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -343,34 +385,50 @@ class WorkspaceMembersPage extends React.Component {
+                                    <Text style={[styles.textLabel, styles.colorMuted]}>
+                                        {this.props.translate('common.noResultsFound')}
+                                    </Text>
+                                </View>
```
@0xmiroslav I have added this condition

PankajAS commented:
@marcaaron @0xmiroslav we are good to merge PR and mark as done now?

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -423,11 +425,13 @@ class WorkspaceMembersPage extends React.Component {
+                                            {this.props.translate('common.noResultsFound')}
+                                        </Text>
+                                    </View>
+)
```
Done @0xmiroslav

PankajAS commented:
@marcaaron if everything looks then can you approve and merge PR and assign job on upwork as well.

PankajAS commented:
> > You can retrigger this bot by commenting recheck in this Pull Request
> 
> @marcaaron can you please retrigger bot? It seems @PankajAS signed CLA but still not applied in workflow

@marcaaron its still not applied in workflow, can you please check

PankajAS commented:
Updated @marcaaron

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -299,6 +328,18 @@ class WorkspaceMembersPage extends React.Component {
+                || this.isKeywordMatch(member.phoneNumber, searchValue)
+                || this.isKeywordMatch(member.firstName, searchValue)
+                || this.isKeywordMatch(member.lastName, searchValue)) {
+                return true;
```
This conditional can get inline instead of explicitly returning true / false.

PankajAS replied:
```suggestion
                data = _.filter(data, member => this.isKeywordMatch(member.displayName, searchValue) || this.isKeywordMatch(member.login, searchValue) || this.isKeywordMatch(member.phoneNumber, searchValue) || this.isKeywordMatch(member.firstName, searchValue) || this.isKeywordMatch(member.lastName, searchValue));
```

@marcaaron i have set condition inline but lint giving warning `This line has a length of 307. Maximum allowed is 190`. if its ok then i can commit this change?

PankajAS replied:
yes @0xmiroslav i try to put them multiple lines but lint don't allow it and error is `error  Trailing spaces not allowed `

PankajAS replied:
@marcaaron if its not necessary can we skip this optimization?

marcaaron replied:
I think we can break it out into multiple lines. If you have trailing spaces that sounds like a different issue.

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -316,7 +357,10 @@ class WorkspaceMembersPage extends React.Component {
                             onCloseButtonPress={() => Navigation.dismissModal()}
-                            onBackButtonPress={() => Navigation.navigate(ROUTES.getWorkspaceInitialRoute(policyID))}
+                            onBackButtonPress={() => {
+                                this.clear();
```
Why is this necessary?

PankajAS replied:
@0xmiroslav @marcaaron  we can keep this?

marcaaron replied:
I'm not following why we need to do anything or what problem it solves.

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -89,6 +92,10 @@ class WorkspaceMembersPage extends React.Component {
         this.getWorkspaceMembers();
     }
 
+    onSearch(searchValue = '') {
```
/**
     * On Search change state of searchValue
     *
     * @param {String} searchValue
 */
     
     @marcaaron that comment looks ok?

PankajAS commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -216,7 +231,7 @@ class WorkspaceMembersPage extends React.Component {
     validate() {
         const errors = {};
-        _.each(this.state.selectedEmployees, (member) => {
+        _.each(this.state.selectedMembers, (member) => {
```
Ok @0xmiroslav @marcaaron

PankajAS commented:
@marcaaron  @0xmiroslav i have commit these feedbacks:
1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

PankajAS commented:
> @PankajAS if feedback done, please mark as resolved per each comment

Done @0xmiroslav

PankajAS commented:
@marcaaron Code is updated, can you please check and we can merge PR today.

PankajAS commented:
Updated @marcaaron , can we merge PR today as functionality works good.

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -89,6 +92,15 @@ class WorkspaceMembersPage extends React.Component {
     }
 
+    /**
+     * On Search change state of searchValue
```
```suggestion
```

This doc comment is not really necessary and should be removed.

PankajAS replied:
@marcaaron you mention in previous suggestion This needs a JSDoc since it has an argument.

marcaaron replied:
Correct. It needs a doc but not a doc with a comment. Thanks!

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -89,6 +92,15 @@ class WorkspaceMembersPage extends React.Component {
+     *
+     * @param {String} searchValue
+     */
+    onSearch(searchValue = '') {
```
as per the style guide methods should be named for what they do e.g. `updateSearchValue()` and not what they handle or the user action taken.

PankajAS replied:
Done @marcaaron

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -227,6 +242,20 @@ class WorkspaceMembersPage extends React.Component {
+    isKeywordMatch(value, keyword) {
+        if (typeof value !== 'string') {
+            return false;
+        }
```
I'm not seeing why it would be `undefined` or `null`.

marcaaron replied:
@PankajAS can you remove this please? It's not needed as far as I can tell.

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -316,7 +359,10 @@ class WorkspaceMembersPage extends React.Component {
                             onCloseButtonPress={() => Navigation.dismissModal()}
-                            onBackButtonPress={() => Navigation.navigate(ROUTES.getWorkspaceInitialRoute(policyID))}
+                            onBackButtonPress={() => {
+                                this.clearMembersAndSearchValue();
```
Why do we need to do this?

PankajAS replied:
@marcaaron its fixing issue state reset when page navigation to back or invite page then come back to Members list page

marcaaron replied:
Yeah I'm not sure we need that. Let's remove it for now? And then also clean up the method called `clearMembersAndSearchValue()`. It's not really needed as a method if we only use it once and we can simplify this code. Thanks!

PankajAS replied:
Ok @marcaaron i will remove it for now

PankajAS commented:
Updated @marcaaron

PankajAS commented:
Update @marcaaron

PankajAS commented:
Updated @marcaaron

PankajAS commented:
@marcaaron everything look ok now? we can merge PR today?

marcaaron commented:
Something is wrong with the CLA assistant. Can you try to add your comment again?

See [this](https://github.com/Expensify/App/pull/15123#issuecomment-1429419545). It has to be the same format.

marcaaron commented on src/pages/workspace/WorkspaceMembersPage.js:
```diff
@@ -227,6 +237,20 @@ class WorkspaceMembersPage extends React.Component {
     }
 
+    /**
+     * Check if value matches with keyword
+     *
```
```suggestion
```

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
> @PankajAS can you please fix all feedback @marcaaron suggested? These are still remaining:


>     * split into multiple lines so fix lint
> 
> 
> ```diff
> -       data = _.filter(data, member => this.isKeywordMatch(member.displayName, searchValue) || this.isKeywordMatch(member.login, searchValue) || this.isKeywordMatch(member.phoneNumber, searchValue) || this.isKeywordMatch(member.firstName, searchValue) || this.isKeywordMatch(member.lastName, searchValue));
> +       data = _.filter(data, member => (
> +           this.isKeywordMatch(member.displayName, searchValue)
> +           || this.isKeywordMatch(member.login, searchValue)
> +           || this.isKeywordMatch(member.phoneNumber, searchValue)
> +           || this.isKeywordMatch(member.firstName, searchValue)
> +           || this.isKeywordMatch(member.lastName, searchValue)
> +       ));
> ```

@0xmiroslav lint throwing error when we set multiple like this `Trailing spaces not allowed`

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
> Something is wrong with the CLA assistant. Can you try to add your comment again?
> 
> See [this](https://github.com/Expensify/App/pull/15123#issuecomment-1429419545). It has to be the same format.

Done @marcaaron , but still its showing error

PankajAS commented:
**recheck**

PankajAS commented:
Updated @marcaaron @0xmiroslav

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
Updated @marcaaron @0xmiroslav , All Feedbacks done, please check

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
@0xmiroslav its not required for sing CLA?

PankajAS commented:
<img width="1053" alt="Screenshot 2023-02-17 at 4 04 33 PM" src="https://user-images.githubusercontent.com/17126628/219620714-e28f0e11-198f-4aed-b5c3-3b62d043a212.png">

PankajAS commented:
> > <img alt="Screenshot 2023-02-17 at 4 04 33 PM" width="1053" src="https://user-images.githubusercontent.com/17126628/219620714-e28f0e11-198f-4aed-b5c3-3b62d043a212.png">
> 
> This should be in your local git. Don't commit to the repo

Ok @0xmiroslav

PankajAS commented:
Updated @marcaaron

PankajAS commented:
Done @0xmiroslav , now latest commits is verified but still CLA failed, do i need to send "_**I have read the CLA Document and I hereby sign the CLA**_" again?

PankajAS commented:
> > Done @0xmiroslav , now latest commits is verified but still CLA failed, do i need to send "_**I have read the CLA Document and I hereby sign the CLA**_" again?
> 
> Let's wait @marcaaron to retrigger

Ok @0xmiroslav

marcaaron commented:
Code looks great to me! Thanks for the reviews @0xmiroslav and to @PankajAS for your patience. Nice first PR 💪 

1. I'm not sure why the CLA check is failing, but I think we can manually skip it since something is messed up with it.
2. Do we need to update the checklist? It looks to be failing now.
3. I just realized we have unsigned commits 🙃 as explained [here](https://github.com/Expensify/App/blob/b113107ba03dce15a4e87bad994829987f916dfa/contributingGuides/CONTRIBUTING.md#begin-coding-your-solution-in-a-pull-request) this step should have been completed before starting.

marcaaron commented:
I personally don't care if we rebase and force push here. It's annoying that it re-writes history but the conversations here were not _that_ interesting :D

marcaaron commented:
@PankajAS feel free to do something like this -> https://github.com/wprig/wprig/wiki/How-to-squash-commits

marcaaron commented:
you'd have to use `git rebase -i main` though

PankajAS commented:
> git rebase -i main

Ok @marcaaron let me do it

PankajAS commented:
Updated @marcaaron , please check now

PankajAS commented:
@marcaaron if its looks ok now then can we merge PR today?

PankajAS commented:
Update @marcaaron, can we check today and merge PR if not everything ok, Thanks

marcaaron commented:
@PankajAS have to squash the commits and then force push in order for my suggestion to work. I'll take a look at this again on Monday. In the future, please do not leave more than one "Updated" comment. It will not expedite the process - we move as fast as we can.

PankajAS commented:
Ok @marcaaron

marcaaron commented:
I'm going OOO for a bit so you will need to find someone to merge this (ask in the open source channel). @PankajAS follow the directions in [this comment](https://github.com/Expensify/App/pull/15123#issuecomment-1435240018).

PankajAS commented:
> I'm going OOO for a bit so you will need to find someone to merge this (ask in the open source channel). @PankajAS follow the directions in [this comment](https://github.com/Expensify/App/pull/15123#issuecomment-1435240018).

@marcaaron i have followed this steps, can you please assign someone who can help to merge if you are going OOO

PankajAS commented:
@0xmiroslav please check now i think it squash all commits into 1 commit

mountiny commented:
@PankajAS you are missing one checkbox in the checklist

mountiny reviewed (approved):
One comment above

mountiny commented:
@PankajAS can you also try to comment the CLA check again please?

PankajAS commented:
Ok @mountiny

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
> I included screenshots or videos for tests on

Updated @mountiny @0xmiroslav

PankajAS commented:
> @PankajAS can you also try to comment the CLA check again please?

Done @mountiny

PankajAS commented:
_**I have read the CLA Document and I hereby sign the CLA**_

PankajAS commented:
@mountiny can we merge PR today?

PankajAS commented:
@mountiny @0xmiroslav can we check if we can merge PR now?

PankajAS commented:
@0xmiroslav i have do anything or wait for it?

PankajAS commented:
Ok @0xmiroslav let me try with closed PR

PankajAS commented:
not able to re-open closed PR @0xmiroslav , can i create new PR and close this?

PankajAS commented:
Ok @0xmiroslav

PankajAS commented:
@0xmiroslav in closed PR there is merge commits and it cant be delete by that branch and my commits are not verified in closed PR, i think we can create new PR and it will be quick, what you think?

PankajAS commented:
> ok let's wait @mountiny to re-trigger here

Ok @0xmiroslav

mountiny commented:
recheck

PankajAS commented:
@mountiny its still CLA failed

mountiny commented:
@PankajAS @0xmiroslav 
<img width="495" alt="image" src="https://user-images.githubusercontent.com/36083550/220676198-884d4fa2-c7b6-4f10-8472-16031d2e1404.png">
seems like I cant

mountiny commented:
@PankajAS Can you create a dummy PR in some of the markdown contributingGuidelines files and sign the CLA there?

PankajAS commented:
@mountiny i need to create new PR?

mountiny commented:
@PankajAS Yes please, I think that worked in some cases in the past

PankajAS commented:
Ok @mountiny so i can close this PR first

mountiny commented:
> Ok @mountiny so i can close this PR first

@PankajAS Nono, Just some dummy change, add a whitespace to end of some line in a markdown file on a new branch and create a new PR for it, sign the CLA there.

mountiny commented:
> PR i have to send for same issue, right? and same details which this PR have

The content of the PR is irrelevant its only to trigger the CLA signature again on a new PR

PankajAS commented:
> > PR i have to send for same issue, right? and same details which this PR have
> 
> The content of the PR is irrelevant its only to trigger the CLA signature again on a new PR

Ok @mountiny got it i'm creating dummy PR

PankajAS commented:
<img width="908" alt="Screenshot 2023-02-22 at 9 32 42 PM" src="https://user-images.githubusercontent.com/17126628/220682730-8524a4f0-445e-4971-8556-add0042a8fc9.png">
 
@mountiny should i delete this content while create PR?

PankajAS commented:
@mountiny I have signed CLA in dummy [PR](https://github.com/Expensify/App/pull/15348)

PankajAS commented:
its done @mountiny

mountiny reviewed (approved):
What a saga, thanks everyone 
<img width="918" alt="image" src="https://user-images.githubusercontent.com/36083550/220687530-4c523129-0724-48a2-9f08-8dfed707bb9a.png">

PankajAS commented:
> @PankajAS Seems like the job 3 is not succeeding, I have retried it, its not related but maybe if it fails again, syncing `main` branch with this might be helpful

Ok @mountiny

mountiny commented:
@PankajAS can you merge in main please?

PankajAS commented:
> @PankajAS can you merge in main please?

do i need to update PR branch with main branch?

PankajAS commented:
this branch is updated @mountiny with `main` branch

mountiny commented:
@PankajAS I think you need to pull the latest main from Expensify/App and sync it with your fork and merge that in. You commit if 2 days old and there were PRs merged into main since then

PankajAS commented:
> @PankajAS I think you need to pull the latest main from Expensify/App and sync it with your fork and merge that in. You commit if 2 days old and there were PRs merged into main since then

Ok let me sync with main from Expensify/App

PankajAS committed:
Add Search feature for Workspace members list

1. Added common isKeywordMatch function
2. Add more fields in filter condition: firstName, lastName
3. Combine all of these into one if statement
4. Use common function like clear()

1. Put this.state.searchValue.trim().toLowerCase() this out of loop
2. Space formate

UI Adjustments

Match "No results found" text with new chat text

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

1. Remove updateSearchValue Comment
2. Update OnSearch name to updateSearchValue
3. Inline conditions for Filter list data

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Miroslav Stevanovic <97473779+0xmiroslav@users.noreply.github.com>

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

PankajAS commented:
Done @mountiny

PankajAS commented:
now we are all set to merge PR? @mountiny @0xmiroslav

mountiny commented:
@PankajAS have to wait for the actions to succeed. Seems like there is something odd going on with the `job3`

PankajAS commented:
Ok @mountiny

mountiny commented:
Looking good

PankajAS commented:
Great @mountiny

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4247587899) to staging by https://github.com/mountiny in version: 1.2.76-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4287101550) to production by https://github.com/yuwenmemon in version: 1.2.76-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4287101550) to production by https://github.com/yuwenmemon in version: 1.2.76-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4287101550) to production by https://github.com/yuwenmemon in version: 1.2.76-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

sobitneupane commented:
Dismissing keyboard on scroll on native devices(Android and IOS) was not considered in this PR. [The issue](https://github.com/Expensify/App/issues/15695) was later reported and solved by [this PR ](https://github.com/Expensify/App/pull/15903)

Steps to reproduce:

- Open manage members page of any workspace on Android.
- Tap on search and scroll through members
- The keyboard is expected to be dismissed (similar to search page, invite member page)

sobitneupane commented:
There was some consistency [issue](https://github.com/Expensify/App/issues/15661) not considered in this PR. Search value was not being cleared while moving between the pages.
